### PR TITLE
Update bson gitignore

### PR DESCRIPTION
**Pull request by stankudrow**
Template: https://github.com/github/gitignore/blob/main/Python.gitignore

Partially covers: #126 - extracting only the `.gitignore` file updates
Partially addresses: #127

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,25 +39,136 @@ pip-delete-this-directory.txt
...
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+.python-version
```
Comment (stankudrow) [inline .gitignore:88]: @Kludex , should it be commented or not? There is an [answer](https://stackoverflow.com/questions/54315206/should-we-gitignore-the-python-version-file), but your decision will be final.

Comment (Kludex) [reply .gitignore:88]: You can add the `.python-version` to the gitignore, yes.

But I don't think we need to add that many things to the `.gitignore`, only add when things are needed.

Comment (stankudrow) [reply .gitignore:88]: Which would you like to keep here definitely? I think I can remove other project management packages, but what else?

File: .gitignore [hunk 2]
```diff
@@ -37,25 +39,68 @@ pip-delete-this-directory.txt
...
+.hypothesis/
+.pytest_cache/
+cover/
 
 # Translations
 *.mo
 *.pot
 
-# Django stuff:
-*.log
```
Comment (stankudrow) [inline .gitignore:52]: to keep or not?

File: .gitignore [hunk 3]
```diff
@@ -8,7 +8,6 @@ __pycache__/
 
 # Distribution / packaging
 .Python
-env/
```
Comment (stankudrow) [inline .gitignore:11]: Moved down to the "Environment" section.

**Comment by stankudrow** [comment]
Rebased.

**Review by Kludex — COMMENTED** [review]
As I said before, I prefer to have files added to the `.gitignore` when needed.

**Pull request closed without merging**